src/sasl/cyrsasl_scram_sha256.erl

```diff
@@ -8,245 +8,12 @@
-    {error, <<"bad-protocol">>}.
-
-check_channel_binding({CBind, _, _ }) when (CBind =/= $n) andalso (CBind =/=$y) ->
-    {error, <<"bad-channel-binding">>};
-check_channel_binding({_CBind, NonceAtt, ClientProofAtt}) ->
-    {ok, {NonceAtt, ClientProofAtt}}.
-
-verify_nonce([{{$r, Nonce}, Nonce} | Args]) ->
-    Args;
-verify_nonce([{{$r, _ClientNonce}, _Nonce} | _Args]) ->
-    {error, <<"bad-nonce">>};
-verify_nonce(_) ->
-    {error, <<"bad-protocol">>}.
-
-get_client_proof([{$p, ClientProofB64}, AuthMessage, State]) ->
-    {jlib:decode_base64(ClientProofB64), AuthMessage, State};
-get_client_proof(_) ->
-    {error,<<"bad-protocol">>}.
-
-get_stored_key({ClientProof, AuthMessage, State}) ->
-    ClientSignature = mongoose_scram_sha256:client_signature(State#state.stored_key,
-                                                      AuthMessage),
-    ClientKey = mongoose_scram_sha256:client_key(ClientProof, ClientSignature),
-    CompareStoredKey = mongoose_scram_sha256:stored_key(ClientKey),
-    {CompareStoredKey, State#state.stored_key}.
-
--spec verify_stored_key({binary(), binary()}) -> {ok, auth_successful} | error().
-verify_stored_key({StoredKey, StoredKey}) ->
-    {ok, auth_successful};
-verify_stored_key(_) ->
-    {error, <<"bad-auth">>}.
-
-%%--------------------------------------------------------------------
-%% Helpers
-%%--------------------------------------------------------------------
-
-parse_attribute(Attribute) when byte_size(Attribute) >= 3 ->
-    parse_attribute(Attribute, 0);
-parse_attribute(_) ->
-    {error, <<"bad-format attribute too short">>}.
-
-parse_attribute(<<Char, Rest/binary>>, 0) ->
-    case is_alpha(Char) of
-        true ->
-            parse_attribute(Rest, Char);
-        _ ->
-            {error, <<"bad-format attribute too short">>}
-    end;
-parse_attribute(<<$=, Rest/binary>>, PName) ->
-    {PName, Rest};
-parse_attribute(_,_) ->
-   {error, <<"bad-format first char not a letter">>}.
-
-unescape_username(<<"">>) -> <<"">>;
-unescape_username(UnescapedUsername) ->
-    case binary:match(UnescapedUsername, <<"=">>) of
-        nomatch ->
-            UnescapedUsername;
-        _ ->
-            EscapedUsername = do_escape_username(UnescapedUsername),
-            check_escaped_username(EscapedUsername)
-    end.
-
-do_escape_username(UnescapedUsername) ->
-    ReplacedEqual = binary:replace(UnescapedUsername, <<"=3D">>, <<"=">>, [global]),
-    binary:replace(ReplacedEqual, <<"=2C">>, <<",">>, [global]).
-
-check_escaped_username(EscapedUsername) ->
-    case binary:match(EscapedUsername, <<"=">>) of
-        nomatch ->
-            EscapedUsername;
-        _ ->
-            error
-    end.
+mech_new(Host, Creds) ->
+    cyrsasl_scram:mech_new(Host, Creds, sha256).
 
-is_alpha(Char) when Char >= $a, Char =< $z -> true;
-is_alpha(Char) when Char >= $A, Char =< $Z -> true;
-is_alpha(_) -> false.
+mech_step(State, ClientIn) ->
+    cyrsasl_scram:mech_step(State, ClientIn).
```

Comment:
EOL in the last line is missing, VSC tends to do that by default 🙂

Reply:
VSCode has a plugin for that https://github.com/riccardoNovaglia/vsCodeBlankLine ;)

Reply:
Thanks @aleklisi, got it enabled.